solarwinds_apm/distro.py

```diff
@@ -213,11 +157,54 @@ def _configure(self, **kwargs):
             OTEL_PYTHON_LOG_FORMAT,
             "%(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] [trace_id=%(otelTraceID)s span_id=%(otelSpanID)s trace_flags=%(otelTraceSampled)02d resource.service.name=%(otelServiceName)s] - %(message)s",
         )
-
         # TODO: Support other signal types when available
         # Always opt into new semconv for all instrumentors (if supported)
         environ["OTEL_SEMCONV_STABILITY_OPT_IN"] = self.get_semconv_opt_in()
 
+        # setdefaults for OTLP export if protocol is unset or valid,
+        # with the exception of OTEL_TRACES_EXPORTER if legacy.
+        # Does not setdefaults if invalid protocol.
+        # All values can still be configured by user (SolarWindsApmConfig).
+        environ.setdefault(
+            OTEL_EXPORTER_OTLP_PROTOCOL,
+            "http/protobuf",
+        )
+        # Protocol is the above default or what user has set
+        otlp_protocol = environ.get(OTEL_EXPORTER_OTLP_PROTOCOL)
+
+        is_legacy = SolarWindsApmConfig.calculate_is_legacy()
+
+        if otlp_protocol:
+            if otlp_protocol in _EXPORTER_BY_OTLP_PROTOCOL:
+                header_token = self._get_token_from_service_key()
+                if not header_token:
+                    logger.debug(
+                        "Setting OTLP export defaults without SWO token"
+                    )
```

Comment:
This is level `debug` because (1) Configurator/ApmConfig will later log `warning` if service key missing or incorrectly formatted when init'ing c-lib, (2) it is valid in Lambda to not have service key and in https://github.com/solarwinds/apm-python/pull/494 I chose to remove the is_lambda check.